internal_api.go

```diff
@@ -310,3 +310,46 @@ func (e *Enforcer) removeFilteredPolicy(sec string, ptype string, fieldIndex int
 
 	return ruleRemoved, nil
 }
+
+func (e *Enforcer) updateFilteredPolicies(sec string, ptype string, oldRules [][]string, newRules [][]string) (bool, error) {
+	if e.dispatcher != nil && e.autoNotifyDispatcher {
+		return true, e.dispatcher.UpdateFilteredPolicies(sec, ptype, oldRules, newRules)
+	}
+
+	if e.shouldPersist() {
+		if err := e.adapter.(persist.UpdatableAdapter).UpdateFilteredPolicies(sec, ptype, oldRules, newRules); err != nil {
+			if err.Error() != notImplemented {
+				return false, err
+			}
+		}
+	}
+
+	ruleRemoved := e.model.RemovePolicies(sec, ptype, oldRules)
```

Comment:
It is necessary to check whether the old rules exist, then remove that.

Reply:
Not exactly, cause this method only called by `UpdateFilteredNamedPolicies` in Enforcer, it's unlikely has excluded rules